test/ptp/ptp.go

```diff
@@ -305,6 +247,133 @@ var _ = Describe("[ptp]", func() {
+		Expect(err).ToNot(HaveOccurred())
+	}
+
+	nodeList, err = client.Client.Nodes().List(metav1.ListOptions{LabelSelector: fmt.Sprintf("%s=", PtpSlaveNodeLabel)})
+	Expect(err).ToNot(HaveOccurred())
+	for _, node := range nodeList.Items {
+		delete(node.Labels, PtpSlaveNodeLabel)
+		_, err = client.Client.Nodes().Update(&node)
+		Expect(err).ToNot(HaveOccurred())
+	}
+
+	ptpNodes, err := nodes.GetNodeTopology(client.Client)
+	Expect(err).ToNot(HaveOccurred())
+	Expect(len(ptpNodes)).To(BeNumerically(">", 1), "need at least two nodes with ptp capable nics")
+
+	By("Labeling the grandmaster node")
+	ptpGrandMasterNode := ptpNodes[0]
+	ptpGrandMasterNode.NodeObject, err = nodes.LabelNode(ptpGrandMasterNode.NodeName, PtpGrandmasterNodeLabel, "")
+	Expect(err).ToNot(HaveOccurred())
+
+	By("Labeling the slave node")
+	ptpSlaveNode := ptpNodes[1]
+	ptpSlaveNode.NodeObject, err = nodes.LabelNode(ptpSlaveNode.NodeName, PtpSlaveNodeLabel, "")
+	Expect(err).ToNot(HaveOccurred())
+
+	By("Creating the policy for the grandmaster node")
+	err = createConfig(PtpGrandMasterPolicyName,
+		ptpGrandMasterNode.InterfaceList[0],
+		"",
+		"-a -r -r",
+		PtpGrandmasterNodeLabel,
+		pointer.Int64Ptr(5))
+	Expect(err).ToNot(HaveOccurred())
+
+	By("Creating the policy for the slave node")
+	err = createConfig(PtpSlavePolicyName,
+		ptpSlaveNode.InterfaceList[0],
+		"-s",
+		"-a -r",
+		PtpSlaveNodeLabel,
+		pointer.Int64Ptr(5))
+	Expect(err).ToNot(HaveOccurred())
+
+	By("Restart the linuxptp-daemon pods")
+	ptpPods, err := client.Client.Pods(PtpLinuxDaemonNamespace).List(metav1.ListOptions{LabelSelector: "app=linuxptp-daemon"})
+	Expect(err).ToNot(HaveOccurred())
+	for _, pod := range ptpPods.Items {
+		err = client.Client.Pods(PtpLinuxDaemonNamespace).Delete(pod.Name, &metav1.DeleteOptions{GracePeriodSeconds: pointer.Int64Ptr(0)})
+		Expect(err).ToNot(HaveOccurred())
+	}
+}
+
+// Returns the slave node label to be used in the test
+func checkPolicies(namespace string) (string, string, string, string) {
+	var masters []ptpv1.PtpConfig
+	var slaves []ptpv1.PtpConfig
+
+	configList, err := client.Client.PtpConfigs(namespace).List(metav1.ListOptions{})
+	Expect(err).ToNot(HaveOccurred())
+	for _, config := range configList.Items {
+		for _, profile := range config.Spec.Profile {
+			if isPtpMaster(*profile.Ptp4lOpts, *profile.Phc2sysOpts) {
+				masters = append(masters, config)
+			}
+			if isPtpSlave(*profile.Ptp4lOpts, *profile.Phc2sysOpts) {
+				slaves = append(masters, config)
+			}
+		}
+	}
+	if len(masters) == 0 {
+		Skip("PTP grandmaster config not found in discovery mode")
+	}
+	if len(slaves) == 0 {
+		Skip("PTP slave config not found in discovery mode")
+	}
+	masterConfigName, masterLabel := checkPtpProfileLabels(masters)
+	slaveConfigName, slaveLabel := checkPtpProfileLabels(slaves)
+	return masterConfigName, slaveConfigName, masterLabel, slaveLabel
+}
+
+func checkPtpProfileLabels(configs []ptpv1.PtpConfig) (string, string) {
```

Comment:
This should be "retrieve" instead of check. Also, `PTP` uppercase as acronyms are all expected to be consistent (see https://github.com/golang/go/wiki/CodeReviewComments#initialisms)

Reply:
Done